assets/src/edit-story/app/media/utils/useUploadVideoFrame.js

```diff
@@ -54,11 +56,15 @@ function useUploadVideoFrame({ updateMediaElement }) {
         featured_media: posterId,
         post: storyId,
       });
-      if (elementId) {
-        // TODO Update all elements that use video.
-        const newState = { resource: { ...resource, posterId, poster } };
-        setProperties(elementId, newState);
-      }
+      const newState = ({ resource, ...rest }) => ({
+        ...rest,
```

Comment:
Top level `...rest` is not needed. Just like your code here, the `setProperties` does `{...element, newProps}`. As a consequence we need to make sure `resource` is fully provided with all merged fields - just like you have now.

Reply:
Yes! Fixed